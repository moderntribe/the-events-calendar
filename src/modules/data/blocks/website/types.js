/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_WEBSITE_URL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_URL`;
