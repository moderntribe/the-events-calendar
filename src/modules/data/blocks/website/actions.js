/**
 * Internal dependencies
 */
import * as types from './types';

export const setWebsite = ( url ) => ( {
	type: types.SET_WEBSITE_URL,
	payload: {
		url,
	},
} );
