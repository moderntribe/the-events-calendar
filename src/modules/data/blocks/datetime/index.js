/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas, setInitialState, defaultStateToMetaMap };
