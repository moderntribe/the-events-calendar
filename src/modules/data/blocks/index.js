/**
 * Internal dependencies
 */
import * as datetime from './datetime';
import subscribe from './subscribers';

export { datetime, subscribe };
export { default, setInitialState } from './reducer';
