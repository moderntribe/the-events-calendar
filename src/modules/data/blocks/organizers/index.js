/**
 * Internal dependencies
 */
import reducer, { setInitialState } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import subscribe from './subscribers';

export default reducer;
export { types, actions, selectors, setInitialState, subscribe };
