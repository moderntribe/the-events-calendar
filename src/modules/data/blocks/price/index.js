/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils, setInitialState, defaultStateToMetaMap };
