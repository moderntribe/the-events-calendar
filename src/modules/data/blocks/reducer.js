/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers, { setInitialState as organizersSetInitialState } from './organizers';
import price, { setInitialState as priceSetInitialState } from './price';
import website, { setInitialState as websiteSetInitialState } from './website';
import venue, { setInitialState as venueSetInitialState } from './venue';

export const setInitialState = ( data ) => {
	datetimeSetInitialState( data );
	organizersSetInitialState( data );
	priceSetInitialState( data );
	venueSetInitialState( data );
	websiteSetInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
