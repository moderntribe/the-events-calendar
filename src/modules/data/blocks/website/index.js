/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types, setInitialState, defaultStateToMetaMap };
