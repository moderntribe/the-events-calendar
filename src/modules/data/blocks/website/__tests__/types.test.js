/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events/data/blocks/website';

describe( '[STORE] - Website types', () => {
	it( 'Should match the types values', () => {
		expect( types.SET_WEBSITE_URL ).toMatchSnapshot();
	} );
} );
