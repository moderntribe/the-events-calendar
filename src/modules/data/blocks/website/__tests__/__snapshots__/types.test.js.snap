// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Website types Should match the types values 1`] = `"@@MT/EVENTS/SET_WEBSITE_URL"`;
